propagation.go

```diff
@@ -178,7 +178,13 @@ func extractTraceContext(opaqueCarrier interface{}) (SpanContext, error) {
 		return spanContext, ot.ErrSpanContextNotFound
 	}
 
-	if !spanContext.Suppressed && (spanContext.SpanID == 0 != (spanContext.TraceIDHi == 0 && spanContext.TraceID == 0)) {
+	// when the context is not suppressed
+	// and instana headers either can not be parsed or present partially
+	// and w3 context is not there
+	if !spanContext.Suppressed &&
+		(spanContext.SpanID == 0 != (spanContext.TraceIDHi == 0 && spanContext.TraceID == 0)) &&
+		spanContext.W3CContext.IsZero() {
```

Comment:
So if there is a non-suppressed request that has a W3C trace and a _broken_ Instana context, we'll return a span context with inconsistent Instana IDs and a valid W3C trace context, right? Since Instana IDs always take precedence when present, we'll use these broken values to continue the trace and probably break it.

I believe in this case we should give up using Instana trace context, probably log this error and continue as if there are no `X-Instana-` headers have been set.

Reply:
Thank you for your feedback, I think your suggestions make sense. I've changed the code.

- Now if only some Instana ID headers present and the w3 context is not zero, I do reset context IDs and continue without error. 
- I also added logging for the mentioned case and in two more places (when parsing headers) to see if there any issue with the header format/value. I am using a debug level to avoid writing a huge amount of logs in case of misconfiguration.